src/main/java/com/finsemble/example/fdc3/JavaFDC3Example.java

```diff
@@ -0,0 +1,185 @@
+import com.finsemble.fdc3.*;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.Arrays;
+import java.util.logging.LogManager;
+import java.util.stream.Collectors;
+
+/**
+ * An example to raise intents within an FDC3 2.0 Desktop Agent.
+ */
+public class JavaFDC3Example extends JFrame {
+
+    // The resource name for the logging config
+    private final static String RESOURCE_NAME_LOGGING_CONFIG = "logging.properties";
+
+    // Configure the logger using "logging.properties"
+    static {
+        try {
+            LogManager.getLogManager().readConfiguration(JavaFDC3Example.class.getClassLoader().getResourceAsStream(RESOURCE_NAME_LOGGING_CONFIG));
+        } catch (Exception exception) {
+            System.err.println("Could not set the logging config from '" + RESOURCE_NAME_LOGGING_CONFIG + "'");
+            exception.printStackTrace(System.err);
+        }
+    }
+
+    // The text field for the ticker name
+    private final JTextField tickerTextField = new JTextField("TSLA");
+    // The button to raise the intent
+    private final JButton raiseIntentButton = new JButton("Raise Chart Intent");
+
+    // The Finsemble connection
+    private Finsemble finsemble;
+
+    // The Desktop Agent
+    private DesktopAgent desktopAgent;
+
+    // Connection listener (for logging)
+    private final ConnectionListener connectionListener = new ConnectionListener() {
+        @Override
+        public void disconnected(ConnectionEventGenerator connectionEventGenerator) {
+            System.err.println("disconnected");
+            setVisible(false);
+            dispose();
+        }
+
+        @Override
+        public void error(ConnectionEventGenerator connectionEventGenerator, Exception e) {
+            System.err.println("error");
+            e.printStackTrace();
+        }
+
+        @Override
+        public void onWindowStateReady(ConnectionEventGenerator connectionEventGenerator) {
+            System.err.println("onWindowStateReady");
+        }
+    };
+
+
+    /**
+     * Constructor for the application class.
+     *
+     * @param args command line arguments
+     * @throws Exception when something goes wrong
+     */
+    public JavaFDC3Example(final String[] args) throws Exception {
+        // Configure Finsemble
+        configureFinsemble(args);
+
+        // Set up the UI
+        setupUI();
+
+        // Connect to Finsemble
+        connectFinsemble();
+    }
+
+    /**
+     * Configures Finsemble. The command line arguments are passed in, the app name is set and a Finsemble
+     * ConnectionListener is added.
+     *
+     * @param args command line arguments
+     */
+    private void configureFinsemble(final String[] args) {
+        finsemble = new Finsemble(Arrays.stream(args).sequential().collect(Collectors.toList()), this);
+        finsemble.setAppName(getClass().getSimpleName());
```

Comment:
Doesn't the app name get set from the config passed in?

Reply:
If it is not launched standalone. I will add logic to only set the name if it is not passed in from the command line.